src/sentry/api/endpoints/project_user_reports.py

```diff
@@ -106,9 +115,14 @@ def post(self, request, project):
             report.event_user_id = euser.id
 
         try:
-            report.group = Group.objects.from_event_id(project, report.event_id)
-        except Group.DoesNotExist:
-            pass
+            event = Event.objects.filter(event_id=report.event_id).select_related('group')[0]
+            report.environment = event.get_environment()
```

Comment:
From looking at this, I _think_ this would raise `Environment.DoesNotExist` if the event isn't tagged.

Reply:
All events should have an environment at this point — the value is set to an empty string if no environment is associated with the event, and is unconditionally used to call `Environment.get_or_create`:
https://github.com/getsentry/sentry/blob/5b3975a5cd26a80c12a5ce480ac296e02a1e4163/src/sentry/event_manager.py#L740-L743

Since there isn't any database level constraint on this, I suppose weird things could happen but in other parts of the application we've been treating it effectively as an invariant that all events have an associated environment.

Reply:
@mattrobenolt and @tkaemming I had initially believed that all the events would have an environment, but then I realized there were test that were failing because this was not the case. After looking at the model I added a try-catch block in the event that Environment does not exist. You happen to be reviewing before I finished writing the code so.... :P